### PR TITLE
Adds light eater interaction to marker beacons

### LINKED ISSUE #40957: Nightmare Light Eater does not break mining Marker Beacons
Issue reported from  Round ID: 95539 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])

Reporting client version: 512

The Nightmare's Light Eater arm blade does not instantly break mining marker beacons like it does all other light sources.

### PR DESCRIPTION
**Pull request by kriskog**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Skoglol
fix: Marker beacons are now disintegrated by nightmares light eater
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)
Fixes #40957 and #40601
Bring the marker beacons in line with other lights, destroying them in the first hit by light eater.

### CODE REVIEW

File: code/modules/mining/equipment/marker_beacons.dm [hunk 1]
```diff
@@ -124,8 +124,16 @@ GLOBAL_LIST_INIT(marker_beacon_colors, list(
 			M.add(1)
 			playsound(src, 'sound/items/deconstruct.ogg', 50, 1)
 			qdel(src)
-	else
-		return ..()
+			return
+	if(istype(I, /obj/item/light_eater))
+		var/ash_type = /obj/effect/decal/cleanable/ash
+		var/obj/effect/decal/cleanable/ash/A = new ash_type(loc)
+		A.desc += "\nLooks like this used to be \an [name] some time ago."
```
Comment (coiax) [inline code/modules/mining/equipment/marker_beacons.dm]: Isn't there a proc for generating ash like this

Comment (kriskog) [reply code/modules/mining/equipment/marker_beacons.dm]: The proc takes items, this is a structure. Therefore, this

File: code/modules/mining/equipment/marker_beacons.dm [hunk 2]
```diff
@@ -124,8 +124,16 @@ GLOBAL_LIST_INIT(marker_beacon_colors, list(
 			M.add(1)
 			playsound(src, 'sound/items/deconstruct.ogg', 50, 1)
 			qdel(src)
-	else
-		return ..()
+			return
+	if(istype(I, /obj/item/light_eater))
+		var/ash_type = /obj/effect/decal/cleanable/ash
+		var/obj/effect/decal/cleanable/ash/A = new ash_type(loc)
```
Comment (Cyberboss) [inline code/modules/mining/equipment/marker_beacons.dm]:
```suggestion
		var/obj/effect/decal/cleanable/ash/A = new /obj/effect/decal/cleanable/ash(drop_location())
```

File: code/modules/mining/equipment/marker_beacons.dm [hunk 3]
```diff
@@ -124,8 +124,16 @@ GLOBAL_LIST_INIT(marker_beacon_colors, list(
 			M.add(1)
 			playsound(src, 'sound/items/deconstruct.ogg', 50, 1)
 			qdel(src)
-	else
-		return ..()
+			return
+	if(istype(I, /obj/item/light_eater))
+		var/ash_type = /obj/effect/decal/cleanable/ash
```
Comment (Cyberboss) [inline code/modules/mining/equipment/marker_beacons.dm]:
```suggestion
```

File: code/modules/mining/equipment/marker_beacons.dm (hunk 1 above)
Comment (Cyberboss) [inline code/modules/mining/equipment/marker_beacons.dm]:
```suggestion
		A.desc += "\nLooks like this used to be \a [src] some time ago."
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by kriskog** [comment]
That is much nicer, thanks Cyberboss

**Review by Cyberboss — APPROVED** [review]

**Comment by coiax** [comment]
Speed merged why?

**Pull request merged**